opm/simulators/wells/RateConverter.hpp

```diff
@@ -496,20 +496,34 @@ namespace Opm {
                     if (hydrocarbonPV > 0.) {
                         auto& attr = attributes_hpv[reg];
                         attr.pv += hydrocarbonPV;
-                        attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
-                        attr.rs += fs.Rs().value() * hydrocarbonPV;
-                        attr.rv += fs.Rv().value() * hydrocarbonPV;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                            attr.rs += fs.Rs().value() * hydrocarbonPV;
+                            attr.rv += fs.Rv().value() * hydrocarbonPV;
```

Comment:
Same comment here as the comments below.

Reply:
Followed your suggestion.